plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcConnectorTest.java

```diff
@@ -132,6 +134,15 @@ protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMapp
         return Optional.of(dataMappingTestSetup);
     }
 
+    @Override
+    protected Optional<SetColumnTypeSetup> filterSetColumnTypesDataProvider(SetColumnTypeSetup setup)
+    {
+        if (setup.sourceValueLiteral().equalsIgnoreCase("'char-to-varchar'")) {
+            return Optional.of(new SetColumnTypeSetup(setup.sourceColumnType(), setup.sourceValueLiteral(), setup.newColumnType(), "'" + padEnd("char-to-varchar", 100, ' ') + "'")); // H2 retains leading spaces when converting char(n) columns to varchar(n)
```

Comment:
leading or trailing?


also, it's the correct behavior, we should fix

```
.add(new SetColumnTypeSetup("char(100)", "'char-to-varchar'", "varchar", "'char-to-varchar'"))
```
instead

let's have this as a prep commit / PR

Reply:
cc @ebyhr